src/app/components/services/querybuilder.service.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+angular.module('transmartBaseUi').factory('QueryBuilderService', ['JSON2XMLService',
+  function(JSON2XMLService) {
+
+  var service = {};
+
+  service.convertCohortFiltersToXML = function(cohortFilters, name) {
+    var xml = JSON2XMLService.json2xml({
+        'query_definition': {
+          'query_name': name,
+          'panel': service.convertCohortFiltersToI2B2Structure(cohortFilters)
+        }
+      });
+    return xml;
+  };
+
+  service.convertCohortFiltersToI2B2Structure = function(cohortFilters) {
+    var panels = [];
+
+    _.each(cohortFilters, function(cohortFilter) {
```

Comment:
I just recently found that they deprecate _.each to _.forEach in latest lodash. Could you probably change this.